src/SFA.DAS.Commitments.Api.IntegrationTests/Tests/WhenSimulatingRealWorldApprenticeshipLoad.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MoreLinq;
+using NUnit.Framework;
+using SFA.DAS.Commitments.Api.Controllers;
+using SFA.DAS.Commitments.Api.DependencyResolution;
+using SFA.DAS.Commitments.Api.IntegrationTests.ApiHost;
+using SFA.DAS.Commitments.Api.IntegrationTests.DatabaseSetup;
+using SFA.DAS.Commitments.Api.IntegrationTests.Helpers;
+
+namespace SFA.DAS.Commitments.Api.IntegrationTests.Tests
+{
+    [TestFixture]
+    public class WhenSimulatingRealWorldApprenticeshipLoad
+    {
+        //todo: better handling of testids
+        [Test]
+        public async Task SimulateSlowdownScenario()
+        {
+            const int numberOfGetApprenticeshipTasks = 30, getApprenticeshipCallsPerTask = 6,
+                numberOfGetApprenticeshipsTasks = 1, getApprenticeshipsCallsPerTask = 3;
+
+            await TestLog.Progress("Starting SimulateSlowdownScenario() Test");
+            await TestLog.Progress("Generating call parameters");
+
+            var getApprenticeshipCallParamsPerTaskTask = GetGetApprenticeshipCallParamsPerTask(numberOfGetApprenticeshipTasks, getApprenticeshipCallsPerTask);
+
+            var employerIdsPerTaskTask = GetGetApprenticeshipsCallParamsPerTask(numberOfGetApprenticeshipsTasks, getApprenticeshipsCallsPerTask);
+            var employerIdsPerTask = await employerIdsPerTaskTask;
+
+            var getApprenticeshipCallParamsPerTask = await getApprenticeshipCallParamsPerTaskTask;
```

Comment:
rather than declaring the emplyoerIdsPerTaskTask above, could it not just be 👍 
'var employerIdsPerTaskTask = GetGetApprenticeshipsCallParamsPerTask(numberOfGetApprenticeshipsTasks, getApprenticeshipsCallsPerTask);'

Reply:
Indeed! This was a hold-over from when the two tasks were passed to Task.WhenAll.
Instead of directly awaiting the task, I've moved the await to its 'just in time' position, just before it's used, so the main thread can potentially do some work while waiting.